tests/Unit/Evolution/Systems/RelativisticEuler/Valencia/Test_ConservativeFromPrimitive.cpp

```diff
@@ -0,0 +1,129 @@
+Scalar<DataType> expected_tilde_tau(
+    const Scalar<DataType>& rest_mass_density,
+    const Scalar<DataType>& specific_internal_energy,
+    const Scalar<DataType>& spatial_velocity_squared,
+    const Scalar<DataType>& lorentz_factor, const Scalar<DataType>& pressure,
+    const Scalar<DataType>& sqrt_det_spatial_metric) noexcept {
+  auto result = make_with_value<Scalar<DataType>>(rest_mass_density, 0.);
+  get(result) = (get(pressure) * get(spatial_velocity_squared) +
+                 (get(lorentz_factor) / (1.0 + get(lorentz_factor)) *
+                      get(spatial_velocity_squared) +
+                  get(specific_internal_energy)) *
+                     get(rest_mass_density)) *
+                square(get(lorentz_factor)) * get(sqrt_det_spatial_metric);
+  return result;
+}
+
+template <typename DataType, size_t Dim>
+tnsr::i<DataType, Dim> expected_tilde_s(
+    const Scalar<DataType>& rest_mass_density,
+    const tnsr::i<DataType, Dim, Frame::Inertial>& spatial_velocity_oneform,
+    const Scalar<DataType>& lorentz_factor,
+    const Scalar<DataType>& specific_enthalpy,
+    const Scalar<DataType>& sqrt_det_spatial_metric) noexcept {
+  auto result = make_with_value<tnsr::i<DataType, Dim>>(rest_mass_density, 0.);
+  for (size_t i = 0; i < Dim; ++i) {
+    result.get(i) = spatial_velocity_oneform.get(i) *
+                    square(get(lorentz_factor)) * get(specific_enthalpy) *
+                    get(rest_mass_density) * get(sqrt_det_spatial_metric);
+  }
+  return result;
+}
+
+template <size_t Dim, typename DataType>
+void test_conservative_to_primitive(
+    const gsl::not_null<std::mt19937*> generator,
+    const gsl::not_null<std::uniform_real_distribution<>*> distribution,
+    const DataType& used_for_size) noexcept {
+  const auto rest_mass_density = make_with_random_values<Scalar<DataType>>(
+      generator, distribution, used_for_size);
+  const auto specific_internal_energy =
+      make_with_random_values<Scalar<DataType>>(generator, distribution,
+                                                used_for_size);
+  const auto spatial_velocity_squared =
+      make_with_random_values<Scalar<DataType>>(generator, distribution,
+                                                used_for_size);
+  const auto lorentz_factor = make_with_random_values<Scalar<DataType>>(
+      generator, distribution, used_for_size);
+  const auto specific_enthalpy = make_with_random_values<Scalar<DataType>>(
+      generator, distribution, used_for_size);
+  const auto pressure = make_with_random_values<Scalar<DataType>>(
+      generator, distribution, used_for_size);
+  const auto sqrt_det_spatial_metric =
+      make_with_random_values<Scalar<DataType>>(generator, distribution,
+                                                used_for_size);
+  const auto spatial_velocity_oneform =
+      make_with_random_values<tnsr::i<DataType, Dim>>(generator, distribution,
+                                                      used_for_size);
```

Comment:
CAPTURE the random values

Reply:
why?  The seed is printed out if the test fails